### PR TITLE
Added feature level test coverage to Registry provider (#2415)

### PR DESCRIPTION
**Pull request by ChrisUbben**
Adding more coverage to issue #2415
- Added more cmdlet test coverage for Registry provider.
- Moved "Verify Remove-ItemProperty Whatif" to the feature level tests

### CODE REVIEW

File: test/powershell/Modules/Microsoft.PowerShell.Management/Registry.Tests.ps1 [hunk 1]
```diff
@@ -131,12 +131,252 @@ Describe "Validate Registry Provider" -Tags @('CI', 'RequireAdminOnWindows') {
...
+            try {
+                $result = Get-ItemProperty -Path ".\*" -Filter "*Test*" -ErrorAction Stop
+                throw "Expected exception not thrown"
+            }
+            catch { $_.FullyQualifiedErrorId | Should Be "NotSupported,Microsoft.PowerShell.Commands.GetItemPropertyCommand" }
+        }
+
+        It "Verify Include" {
+            $result = Get-ItemProperty -Path ".\*" -Include "*2*"
+            $result."$testPropertyName" | Should Be $testPropertyValue
```
Comment (mirichmo) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Registry.Tests.ps1]: Include and Exclude should be verified in a measurable way here as well.

File: test/powershell/Modules/Microsoft.PowerShell.Management/Registry.Tests.ps1 [hunk 2]
```diff
@@ -131,12 +131,252 @@ Describe "Validate Registry Provider" -Tags @('CI', 'RequireAdminOnWindows') {
...
+            $result = Get-ItemProperty -Path $testSubKey -Name $testPropertyName
+            $result."$testPropertyName" | Should Be $newPropertyValue
+        }
+
+        It "Verfiy PassThru" {
+            $result = Set-ItemProperty -Path $testSubKey -Name $testPropertyName -Value $newPropertyValue -PassThru
+            $result."$testPropertyName" | Should Be $newPropertyValue
+        }
+
+        It "Verfiy Piped Default Parameter" {
```
Comment (mirichmo) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Registry.Tests.ps1]: Spelling. Verify

File: test/powershell/Modules/Microsoft.PowerShell.Management/Registry.Tests.ps1 [hunk 3]
```diff
@@ -131,12 +131,252 @@ Describe "Validate Registry Provider" -Tags @('CI', 'RequireAdminOnWindows') {
...
+            $result."$testPropertyName" | Should Be $newPropertyValue
+        }
+
+        It "Verfiy WhatIf" {
+            $result = Set-ItemProperty -Path $testSubKey -Name $testPropertyName -Value $newPropertyValue -PassThru -WhatIf
+            $result = Get-ItemProperty -Path $testSubKey -Name $testPropertyName
+            $result."$testPropertyName" | Should Be $testPropertyValue
+        }
+
+        It "Verfiy Confirm" {
```
Comment (mirichmo) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Registry.Tests.ps1]: Bypass

File: test/powershell/Modules/Microsoft.PowerShell.Management/Registry.Tests.ps1 [hunk 4]
```diff
@@ -131,12 +131,252 @@ Describe "Validate Registry Provider" -Tags @('CI', 'RequireAdminOnWindows') {
...
+        }
+    }
+
+    Context "Valdiate Copy-ItemProperty Parameters" {
+        It "Verfiy PassThru" {
+            $result = Copy-ItemProperty -Path $testSubKey -Name $testPropertyName -Destination $testSubKey2 -PassThru
+            $result."$testPropertyName" | Should Be $testPropertyValue
+        }
+
+        It "Verfiy Confirm" {
```
Comment (mirichmo) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Registry.Tests.ps1]: Bypass

**Review by mirichmo — CHANGES_REQUESTED** [review]

File: test/powershell/Modules/Microsoft.PowerShell.Management/Registry.Tests.ps1 [hunk 5]
```diff
@@ -131,12 +131,252 @@ Describe "Validate Registry Provider" -Tags @('CI', 'RequireAdminOnWindows') {
...
+            try {
+                $result = New-ItemProperty -Path ".\*" -Filter "*Test*" -Name $testPropertyName -Value $testPropertyValue -PropertyType DWORD -ErrorAction Stop
+                throw "Expected exception not thrown"
+            }
+            catch { $_.FullyQualifiedErrorId | Should Be "NotSupported,Microsoft.PowerShell.Commands.NewItemPropertyCommand" }
+        }
+
+        It "Verify Include" {
+            $result = New-ItemProperty -Path ".\*" -Include "*2" -Name $testPropertyName -Value $testPropertyValue -PropertyType DWORD
+            $result."$testPropertyName" | Should Be $testPropertyValue
```
Comment (ChrisUbben) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Registry.Tests.ps1]: TL;DR Fixed.
You are right, I applied a path validation to all of the tests that I think it applies. This also found errors in my code so there are refactors in those areas as well.

File: test/powershell/Modules/Microsoft.PowerShell.Management/Registry.Tests.ps1 [hunk 6]
```diff
@@ -131,12 +131,252 @@ Describe "Validate Registry Provider" -Tags @('CI', 'RequireAdminOnWindows') {
...
+        }
+
+        It "Verify Confirm can be bypassed" {
+            $result = New-ItemProperty -Path $testSubKey -Name $testPropertyName -Value $testPropertyValue -PropertyType DWORD -force -Confirm:$false
+            $result."$testPropertyName" | Should Be $testPropertyValue
+        }
+
+        It "Verify WhatIf" {
+            $result = New-ItemProperty -Path $testSubKey -Name $testPropertyName -Value $testPropertyValue -PropertyType DWORD -whatif
+            $result."$testPropertyName" | Should BeNullOrEmpty
```
Comment (ChrisUbben) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Registry.Tests.ps1]: Fixed.

File: test/powershell/Modules/Microsoft.PowerShell.Management/Registry.Tests.ps1 (hunk 1 above)
Comment (ChrisUbben) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Registry.Tests.ps1]: Fixed.

File: test/powershell/Modules/Microsoft.PowerShell.Management/Registry.Tests.ps1 (hunk 3 above)
Comment (ChrisUbben) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Registry.Tests.ps1]: Fixed.

File: test/powershell/Modules/Microsoft.PowerShell.Management/Registry.Tests.ps1 (hunk 4 above)
Comment (ChrisUbben) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Registry.Tests.ps1]: Fixed.

**Review by mirichmo — APPROVED** [review]

**Pull request merged**